forem/forem
# Enable paranoid mode for Devise

## What type of PR is this? (check all applicable)


## Description

This PR enables Devise's "paranoid mode" to prevent email enumeration. From the initializer comment:

>\# It will change confirmation, password recovery and other workflows
>\# to behave the same regardless if the e-mail provided was right or wrong.

This means if a user enters an email address in the forgot password form (or similar forms), they will no longer get any indication if the address they entered actually exists in the system.

## Related Tickets & Documents

Closes https://github.com/forem/InternalProjectPlanning/issues/180

## QA Instructions, Screenshots, Recordings

Automatic:
* Disable paranoid mode in `config/initialzers/devise.rb` and see the spec fail.

Manual:
* Perform the following specs as outlined in the system spec 
    ```ruby
    visit new_user_session_path
    click_link "I forgot my password"
    fill_in "Email", with: "doesnotexist@example.com"
    click_button "Send me reset password instructions"
    ```
* Do this both with paranoid mode enabled and disabled.

## Added tests?

- 
## Added to documentation?
